### ISSUE TITLE
Shrike[build] should allow for hyperdrive/sweep yaml specs living next to component spec without additional includes

### ISSUE THREAD

**Issue by jfomhover** [Azure/shrike#311]
**Category**
Please select:
- [ ] shrike.pipeline
- [ ] shrike.build
- [ ] shrike.compliant_logging

Please also add corresponding labels.

**Is your feature request related to a problem? Please describe.**
When writing a hyperdrive/sweep component, the usual practice would be to place the component yaml next to the component spec. For instance:
- /foo/component_spec.yaml
- /foo/component_spec.additional_includes
- /foo/hyperdrive_spec.yaml

The problem arise during build.prepare when checking [check_for_wrongly_named_additional_includes(component)](https://github.com/Azure/shrike/blob/2c82d09afc37df3ff860e6fc652d45a9200a1af5/shrike/build/commands/prepare.py#L538), it will fail for the hyperdrive_spec.yaml because the component_spec.additional_includes lives next to it.

**Describe the solution you'd like**
Two options:
1. Create a flag in the config file to disable this test (default would be enabled to be backcompatible)
2. Make this test aware on the type of component, and not check additional includes for hyperdrive components

**Estimate the Impact**
1. At least one internal team has this issue.

**Comment by dkmiller** [comment]
Good catch @jfomhover ! My suggestion: rather than making people "flip" a flag, make the "check for ..." method a bit smarter to do the correct thing when there is a sweep component in the same directory.

@fuhuifang , what do you think?
